pkg/reconciler/taskrun/controller.go

```diff
@@ -31,50 +33,57 @@ import (
 	_ "github.com/tektoncd/chains/pkg/chains/formats/all"
 )
 
-func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
-	logger := logging.FromContext(ctx)
-	taskRunInformer := taskruninformer.Get(ctx)
+// NewNamespacesScoppedController returns a new controller implementation where informer is filtered
+// given a list of namespaces
+func NewNamespacesScoppedController(namespaces []string) func(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	return func(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+		logger := logging.FromContext(ctx)
+		taskRunInformer := taskruninformer.Get(ctx)
 
-	kubeClient := kubeclient.Get(ctx)
-	pipelineClient := pipelineclient.Get(ctx)
+		kubeClient := kubeclient.Get(ctx)
+		pipelineClient := pipelineclient.Get(ctx)
 
-	tsSigner := &chains.ObjectSigner{
-		SecretPath:        SecretPath,
-		Pipelineclientset: pipelineClient,
-		Recorder:          taskrunmetrics.Get(ctx),
-	}
+		tsSigner := &chains.ObjectSigner{
+			SecretPath:        SecretPath,
+			Pipelineclientset: pipelineClient,
+			Recorder:          taskrunmetrics.Get(ctx),
+		}
 
-	c := &Reconciler{
-		TaskRunSigner:     tsSigner,
-		Pipelineclientset: pipelineClient,
-	}
-	impl := taskrunreconciler.NewImpl(ctx, c, func(impl *controller.Impl) controller.Options {
-		cfgStore := config.NewConfigStore(logger, func(name string, value interface{}) {
```

Comment:
I don't understand how lint was not failing before here 🤔

Reply:
Lint is configured to only look at new changes, so anything that was failing lint before it was enabled doesn't get flagged until it's modified.